frontend/src/metabase/components/OnClickOutsideWrapper.jsx

```diff
@@ -39,11 +39,14 @@ export default class OnClickOutsideWrapper extends Component {
         window.removeEventListener("click", this._handleClick, true);
         clearTimeout(this._timeout);
 
-        // remove popover from the stack
-        var index = popoverStack.indexOf(this);
-        if (index >= 0) {
-            popoverStack.splice(index, 1);
-        }
+        // remove from the stack after a delay, if it is removed through some other
+        // means this will happen too early causing parent modal to close
+        setTimeout(() => {
+            var index = popoverStack.indexOf(this);
+            if (index >= 0) {
+                popoverStack.splice(index, 1);
```

Comment:
That's a possibility, but I don't think pure UI component state really needs to go in the Redux store. The one disadvantage is in certain cases Redux "time travel" etc won't work correctly, but we have *tons* of component state for things that actually definitely should be in the Redux store, not to mention our over-usage of thunk actions, so time travel is a long ways off from actually working in Metabase anyways.